tests/std/tests/VSO_0000000_regex_use/test.cpp

```diff
@@ -1698,6 +1698,190 @@ void test_gh_5377() {
+            // leftmost-longest rule according to Boost semantics
+            repeat_twice_regex_ere.should_search_match_capture_groups("a", "a", match_default, {{0, 1}});
+        }
+    }
+
+    for (string pattern : {R"(\(a\{0,1\}\)\{2\})", R"(\(a*\)\{2\})"}) {
+        for (auto option : {basic, grep}) {
+            test_regex repeat_twice_regex_bre(&g_regexTester, pattern, option);
+            repeat_twice_regex_bre.should_search_match_capture_groups("", "", match_default, {{0, 0}});
+            repeat_twice_regex_bre.should_search_match_capture_groups("b", "", match_default, {{0, 0}});
+            // leftmost-longest rule according to Boost semantics
+            repeat_twice_regex_bre.should_search_match_capture_groups("a", "a", match_default, {{0, 1}});
+        }
+    }
+
+    for (string pattern : {"(a?a?){2}", "(a?a?)+"}) {
+        test_regex repeated_double_question_regex_ecma(&g_regexTester, pattern, ECMAScript);
+        repeated_double_question_regex_ecma.should_search_match_capture_groups("", "", match_default, {{0, 0}});
+        repeated_double_question_regex_ecma.should_search_match_capture_groups("bbb", "", match_default, {{0, 0}});
+        repeated_double_question_regex_ecma.should_search_match_capture_groups("aaa", "aaa", match_default, {{2, 3}});
+
+        for (auto option : {extended, egrep, awk}) {
+            test_regex repeated_double_question_regex_ere(&g_regexTester, pattern, option);
+            repeated_double_question_regex_ere.should_search_match_capture_groups("", "", match_default, {{0, 0}});
+            repeated_double_question_regex_ere.should_search_match_capture_groups("bbb", "", match_default, {{0, 0}});
+            // leftmost-longest rule according to Boost semantics
+            repeated_double_question_regex_ere.should_search_match_capture_groups(
+                "aaa", "aaa", match_default, {{1, 3}});
+        }
+    }
+
+    for (string pattern : {R"(\(a\{0,1\}a\{0,1\}\)\{2\})", R"(\(a\{0,1\}a\{0,1\}\)\{1,\})"}) {
+        for (auto option : {basic, grep}) {
+            test_regex repeated_double_question_regex_bre(&g_regexTester, pattern, option);
+            repeated_double_question_regex_bre.should_search_match_capture_groups("", "", match_default, {{0, 0}});
+            repeated_double_question_regex_bre.should_search_match_capture_groups("bbb", "", match_default, {{0, 0}});
+            // leftmost-longest rule according to Boost semantics
+            repeated_double_question_regex_bre.should_search_match_capture_groups(
+                "aaa", "aaa", match_default, {{1, 3}});
+        }
+    }
+
+    {
+        test_regex backref_ecma(&g_regexTester, R"(a(b?)+c\1d)", ECMAScript);
+        backref_ecma.should_search_fail("abcd");
+    }
+
+    for (auto option : {basic, grep}) {
+        test_regex backref_bre(&g_regexTester, R"(a\(b?\)\{1,\}c\1d)", option);
+        backref_bre.should_search_fail("abcd");

```

Comment:
Bug: `?` isn't special for BREs, so this literal character will always cause the search to fail. This would have been noticed if there were both positive and negative tests.